overlord/devicestate/devicemgr.go

```diff
@@ -148,31 +151,82 @@ func (m *DeviceManager) ensureOperational() error {
 		return nil
 	}
 
+	// conditions to trigger device registration
+	//
+	// * have a model assertion with a gadget (core and
+	//   device-like classic) in which case we need also to wait
```

Comment:
This one we'll need to rethink at some point. We should be able to ship general classic with gadget snaps and without triggering this logic. The gadget is just a way to customize the behavior of the given model, so it's unrelated to whether the device is registered or not. I get that device-like things should registered automatically, though. We just need to find another delta between those.

Reply:
I can add a comment about this, but if we have a gadget then we have some snaps, so we would need more logic below to not count core and gadet?, also likely we will need a serial to refresh that gadget in some scenarios. I fear we might need the gadget telling us whether to wait or not.

Reply:
Seems okay to postpone these details and keep implementation as-is, as we don't really need the feature right now, but we should at least acknowledge that as a short coming we might eventually fix rather than a plan which we'll be wondering about in the future.